### ISSUE TITLE
Pi interface board

### ISSUE THREAD

**Issue by M0LTE** [M0LTE/pi-transceiver#46]
Suggested 40 pin header pinout: [link](https://docs.google.com/spreadsheets/d/1Nf7KcFOCoI9gorfEW-V2X97y0xYmrL0twZed37Yi8l4/edit?usp=sharing)

Terminals on this board:

```
power
  - gnd                          pins 6, 9, 14, 20, 25, 30, 34, 39 (use multiple)
  - +5v input                    pins 2, 4 (use both)

receiver board
  - gnd
  - tx gpio (gpio 18)            pin 12
  - att1 gpio (gpio 17)          pin 11
  - att2 gpio (gpio 27)          pin 13
  - rx overload (gpio 22)        pin 15
  
pa board
  - gnd
  - i2c sda                      pin 3
  - i2c scl                      pin 5
  - interrupt (gpio 23)          pin 16

psu
  - gnd
  - io 1                         pin 36
  - io 2                         pin 38
  - io 3                         pin 40

misc / spare
  - multiple i2c ports, consisting of:
    - gnd
    - sda                        pin 3
    - scl                        pin 5
  - uart
    - gnd
    - txd                        pin 8
    - rxd                        pin 10
  - 1wire bus (temp sensing, DS18B20)
    - gnd
    - data                       pin 37
    - 3.3V
    - 4.7k pull-up resistor between data and +3.3V
  - multiple 3.3v connections    pins 1, 17
  - multiple 5v connections
  - multiple gnd connections
```

ADC
We have four channels with one module. Probably enough.
makes sense to have four pairs of pins of:
 - gnd
 - analogue V in

We could solder a pre-made ADS1115 module on to a header on this hat, or we could
go chip-level and put a few components around it.
The ADS1115 needs to connect to the I2C pins on the 40 pin header.
My assumption is with such short wires we can have a star topology I2C bus and don't need to loop
the SDA and SCL pins out and back, or worse, rats nest between all the boards.

DAC
This is for an analogue meter etc.
@Simon - this interface is down to you. Did you want to put this at the other end of an I2C interface?
Alternatively, we could do hardware PWM driving a moving meter coil presumably via a transistor. You decide.
If we did hardware PWM as per the fans below, you could just crimp a standard fan connector on to wires
from the moving coil meter, and we choose in software whether we want two independent fans, or one fan + PWM meter.

Fans:
I suggest we use PWM-control fans with rpm feedback to a GPIO, like this:
https://blog.driftking.tw/en/2019/11/Using-Raspberry-Pi-to-Control-a-PWM-Fan-and-Monitor-its-Speed/
we have two hardware PWM ports on the pi - PWM0 (pin 12 and 32), PWM1 (pin 33 and 35)
I suggest two fan ports using the standard 4 pin PWM fan connector
connections:
  - +5V
  - gnd
  - PWM output, to PWM0 (pin 12) or PWM1 (pin 13)
  - RPM sense, to gpio 5 (pin 29) or gpio 6 (pin 31)
or one PWM port for a fan, and one PWM port for the moving coil meter. (Should we buffer the PWM pins just in case?)

Connectors:
Would like to see 2.54mm pitch connectors for anything low current, and 5mm pitch holes for anything higher current, e.g. power in. Various connectors will fit - [I like these](https://uk.farnell.com/wago/2604-1102/tb-wire-to-board-2pos-24-12awg/dp/2821984)

**Comment by gblades** [comment]
Psu board connections would be :-
Io1 - output, high to switch off the 12v input.
Io2 - input, goes active high if the supply to the 5v dc-dc converter drops below 9V. Used to initiate a shutdown.
Io3 - output, high to indicate the 12v battery backup is to shut off its output.

The normal shutdown sequence would be to set io1 high and wait a second. If there is no battery backup this would cut the power immediately. Otherwise after a second set io3 high to switch off the battery output to avoid it being drained by the pi sitting in a shutdown state.

**Comment by G4CDF** [comment]
Some PCBs will need Power input but at low current eg 100mA.   Should power always use a power connector or is it acceptable to use the signal connectors?  What happens if there is a short circuit ie what current limits are provided?  To complicate matters even more what if the PSU module uses a high current connector and the (say) Receiver module uses a low currect connector what size of connecting wire can be used.  Can multiple wires be easily used in a power socket pin?

**Comment by M0LTE** [comment]
I don't think we can or should protect every module from every other module. I don't think that's how commercial radios are designed is it?

> Can multiple wires be easily used in a power socket pin?
I don't see any need to do this - we'll just put enough terminals on the edge of the board for our needs :-)

**Comment by G4CDF** [comment]
Sounds good.  The Receiver module will need +12V, +5V and gnd power inputs.

**Comment by borzwatson** [comment]
Thanks for all this, I've started doing a board,added to GitHub. Do feel free to dive in, I''ll make sure I pull all the files in before I edit any more when I get time.
I would prefer to use the I2C DA which I will add to the board. It can just be left off by those not interested in these functionality.
Ok on the fan functionality.
Hopefully will get some time to do a bit more soon.

**Comment by borzwatson** [comment]
@M0LTE What is the ADC for please? Since the PA board is i2c I would have thought SWR/Output sensing etc.  and the ADC would be on that board.
The reason I ask is I think the analog input wiring between the ADC and whatever it is measuring needs to be kept short I am not sure what the ADC is going to be connected to. 73 S

**Comment by G4CDF** [comment]
Originally the ADC was for the Power amplifier directional coupler signals to measure forward and reverse power.  However I think that Richard is planning to put the ADC on the PA module.  Richard please confirm.  Other uses would be to monitor voltage rails etc

**Comment by M0LTE** [comment]
Mike is right. One less thing to worry about :)

**Comment by ribbotson** [comment]
Closing this issue as first pass the Pi interface board is designed and built.
